prql-compiler/tests/integration/main.rs

```diff
@@ -30,258 +53,127 @@ mod tests {
+                    continue;
+                }
+                results.insert(vendor, run_query(con.as_mut(), prql.as_str(), &runtime));
+            }
 
-            // save both csv files as same snapshot
-            let opts = Options::default()
-                .with_target(Target::Sql(Some(Dialect::DuckDb)))
-                .no_format();
-            let sql = prql_compiler::compile(&prql, &opts).unwrap();
-            let duckdb_out = duckdb::query_csv(&duckdb_conn, &sql);
-            pretty_assertions::assert_eq!(sqlite_out, duckdb_out, "SQLite == DuckDB: {test_name}");
+            if results.is_empty() {
+                return;
+            }
 
-            if let Some(pg_client) = &mut pg_client {
-                let opts = Options::default()
-                    .with_target(Target::Sql(Some(Dialect::PostgreSql)))
-                    .no_format();
-                let sql = prql_compiler::compile(&prql, &opts).unwrap();
-                let pg_out = postgres::query_csv(pg_client, &sql);
-                pretty_assertions::assert_eq!(sqlite_out, pg_out, "SQLite == PG: {test_name}");
+            let first_result = match results.iter().next() {
+                Some(v) => v,
+                None => return,
+            };
+            for (k, v) in results.iter().skip(1) {
+                pretty_assertions::assert_eq!(
+                    *first_result.1,
+                    *v,
+                    "{} == {}: {test_name}",
+                    first_result.0,
+                    k
+                );
             }
 
-            assert_snapshot!(sqlite_out);
+            let mut result_string = String::new();
+            for row in first_result.1 {
+                writeln!(&mut result_string, "{}", row.join(",")).unwrap_or_default();
+            }
+            assert_snapshot!(result_string);
         });
     }
 
-    /// Return a path relative to the root `integration` path.
-    fn path(relative_path: &str) -> String {
-        // Insired by insta's approach to finding a file in a test path.
-        let root = env!("CARGO_MANIFEST_DIR");
-        Path::new(root)
-            .join("tests/integration")
-            .join(relative_path)
-            .canonicalize()
-            .or_else(|_| Path::new("./").join(relative_path).canonicalize())
-            .unwrap()
-            .to_str()
-            .unwrap()
-            .to_owned()
-    }
-
-    fn load_schema() -> String {
-        fs::read_to_string(path("data/chinook/schema.sql")).unwrap()
-    }
-
-    mod sqlite {
-        use super::path;
-        use rusqlite::{types::ValueRef, Connection};
-
-        pub fn connect() -> Connection {
-            Connection::open(path("data/chinook/chinook.db")).unwrap()
+    fn get_connections(runtime: &Runtime) -> Vec<Box<dyn DBConnection>> {
+        let mut connections: Vec<Box<dyn DBConnection>> = vec![];
+        if let Ok(connection) = duckdb::Connection::open_in_memory() {
```

Comment:
I'm not so sure about this design — if something can't be enabled it'll pass the tests, but actually we want to fail...

Reply:
You're right. That makes sense for in memory dbs